components/org.wso2.carbon.identity.discovery/src/main/java/org/wso2/carbon/identity/discovery/builders/ProviderConfigBuilder.java

```diff
@@ -140,6 +143,23 @@ public OIDProviderConfigResponse buildOIDProviderConfig(OIDProviderRequest reque
         if (OAuth2Util.getSupportedGrantTypes().contains(DEVICE_FLOW_GRANT_TYPE)) {
             providerConfig.setDeviceAuthorizationEndpoint(OAuth2Util.OAuthURL.getDeviceAuthzEPUrl());
         }
+
+        List<String> supportedTokenEndpointSigningAlgorithms = OAuthServerConfiguration.getInstance()
+                .getSupportedTokenEndpointSigningAlgorithms();
+        String[] supportedTokenEndpointSigningAlgorithmsArray = new String[supportedTokenEndpointSigningAlgorithms
+                .size()];
+        try {
+            for (int count = 0; count < supportedTokenEndpointSigningAlgorithms.size(); count++) {
+                supportedTokenEndpointSigningAlgorithmsArray[count] =  String.valueOf(OAuth2Util
+                        .mapSignatureAlgorithmForJWSAlgorithm(supportedTokenEndpointSigningAlgorithms.get(count)));
+            }
+        } catch (IdentityOAuth2Exception e) {
+            throw new ServerConfigurationException("Unsupported signature algorithm configured.", e);
+        }
+        providerConfig.setTokenEndpointAuthSigningAlgValuesSupported(supportedTokenEndpointSigningAlgorithmsArray);
```

Comment:
Improve readability

```suggestion
        List<String> mappedAlgorithms = new ArrayList<>();
try {
    for (String algorithm : supportedTokenEndpointSigningAlgorithms) {
        String convertedAlgorithm = String.valueOf(OAuth2Util
                .mapSignatureAlgorithmForJWSAlgorithm(algorithm));
        mappedAlgorithms.add(convertedAlgorithm);
    }
} catch (IdentityOAuth2Exception e) {
    throw new ServerConfigurationException("Unsupported signature algorithm configured.", e);
}
providerConfig.setTokenEndpointAuthSigningAlgValuesSupported(mappedAlgorithms.toArray(new String[0]));
```

Reply:
Fixed in a0e00c2b1fae6d3e771343b8dcc8840d1787da91